tests/RPCClient.test.ts

```diff
@@ -1179,5 +1179,77 @@ describe(`${RPCClient.name}`, () => {
       },
       { numRuns: 1 },
     );
+    testProp(
+      'handler overrides client timeout - lesser value',
+      [specificMessageArb],
+      async (messages) => {
+        const inputStream = rpcTestUtils.messagesToReadableStream(messages);
+        const [outputResult, outputStream] =
+          rpcTestUtils.streamToArray<Uint8Array>();
+        const streamPair: RPCStream<Uint8Array, Uint8Array> = {
+          cancel: () => {},
+          meta: undefined,
+          readable: inputStream,
+          writable: outputStream,
+        };
+        const ctxProm = promise<ContextTimed>();
+        const rpcClient = new RPCClient({
+          manifest: {},
+          streamFactory: async (ctx) => {
+            ctxProm.resolveP(ctx);
+            return streamPair;
+          },
+          logger,
+          idGen,
+          // Setting timeout here to 150ms
+          streamKeepAliveTimeoutTime: 150,
+        });
+
+        const callerInterface = await rpcClient.duplexStreamCaller<
+          JSONValue,
+          JSONValue
+        >(methodName, { timer: 100 });
+
+        const ctx = await ctxProm.p;
+        expect(ctx.timer.delay).toEqual(100);
+      },
+      { numRuns: 5 },
+    );
+    testProp(
+      'handler overrides client timeout - greater value',
+      [specificMessageArb],
+      async (messages) => {
+        const inputStream = rpcTestUtils.messagesToReadableStream(messages);
+        const [outputResult, outputStream] =
+          rpcTestUtils.streamToArray<Uint8Array>();
+        const streamPair: RPCStream<Uint8Array, Uint8Array> = {
+          cancel: () => {},
+          meta: undefined,
+          readable: inputStream,
+          writable: outputStream,
+        };
+        const ctxProm = promise<ContextTimed>();
+        const rpcClient = new RPCClient({
+          manifest: {},
+          streamFactory: async (ctx) => {
+            ctxProm.resolveP(ctx);
+            return streamPair;
+          },
+          logger,
+          idGen,
+          // Setting timeout here to 150ms
+          streamKeepAliveTimeoutTime: 150,
+        });
+
+        const callerInterface = await rpcClient.duplexStreamCaller<
+          JSONValue,
+          JSONValue
+        >(methodName, { timer: 300 });
+
+        const ctx = await ctxProm.p;
+        expect(ctx.timer.delay).toEqual(300);
+      },
+      { numRuns: 5 },
+    );
```

Comment:
Same here, handler isn't involved so it;s not updating the timeout.

Are you testing that the timeout provided to the client call supersedes the default client timeout?